app/services/dependents_creation_service.rb

```diff
@@ -0,0 +1,62 @@
+class DependentsCreationService
```

Comment:
I think this class needs refactoring. There are two issues that I think need addressing:

- the error structure is too complicated, 
- and the main action of the object (creating dependents) is only triggered if the `errors` method is called.

So I'd suggest. Create a single error store, and a method to add to it easily.
```ruby
attr_reader :error_store

def add_error(error)
  @errors_store ||= []
  @errors_store << error
end
```
Then everywhere an error can be detected:
```ruby
add_error(something.error) if something.error.present?
```

Then make `create_dependents` public and memoize it so it can be called multiple times - but with only the first one having to do the processing. Then call it at the beginning of every process that relies on it having been run. So:

```ruby
def errors
  create_dependents
  error_store.flatten
end
```
That way other devs don't have to dig through the code to work out where the creation is triggered.